examples/expressions/12_scikit_learn_estimators_in_expressions.py

```diff
@@ -1,292 +0,0 @@
-# computation and use ``mark_as_X()`` and ``mark_as_y()`` as soon as possible
-# -- as soon as we have separate ``X`` and ``y`` tables that are aligned and
-# have one row per sample. In particular, we should use ``mark_as_X()`` before
-# doing any feature extraction and selection.
-#
-# Skrub will let us apply transformers before reaching ``mark_as_X`` because
-# there are special cases where we know it is safe and faster to do so, but in
-# general we should be careful and remember to separate features and targets
-# and then use ``mark_as_X()`` as soon as possible.
-
-
-# %%
-# A full example
-# --------------
-#
-# Now we know enough about skrub expressions to create a full estimator for a
-# realistic example and evaluate its performance.
-#
-# We finally come back to the credit fraud dataset from the previous example.
-
-# %%
-import skrub.datasets
-
-dataset = skrub.datasets.fetch_credit_fraud()
-
-# %%
-# The ``baskets`` are orders on a e-commerce website. Note that in the drop-down
-# menu, ``"baskets"`` is both an input variable and a table that is
-# marked as ``X``, and therefore it is colored blue and has a double box.
-
-# %%
-baskets = skrub.var("baskets", dataset.baskets[["ID"]]).skb.mark_as_X()
-baskets
-
-# %%
-# Each basket is associated with a fraud flag. The ``"fraud_flag"`` column in the
-# ``baskets`` table is red as it is marked as ``y``.
-
-# %%
-fraud_flags = skrub.var("fraud_flags", dataset.baskets["fraud_flag"]).skb.mark_as_y()
-fraud_flags
-
-# %%
-# Each basket contains one or several products. The ``"basket_ID"`` column
-# in the ``"products"`` refers to the ``"ID"`` column in the ``"baskets"``
-# table. Given the information we have about a basket, we want to predict if
-# its purchase was fraudulent.
-
-# %%
-products = skrub.var("products", dataset.products)
-products
-
-# %%
-# The ``"baskets"`` table itself contains no
-# information: we have to bring in features extracted from the corresponding
-# products. To do so, we must first vectorize the products lines, so that we
-# can aggregate the extracted numeric features, and attach the resulting vector
-# to the corresponding ``"basket_ID"`` and thus the corresponding fraud flag.
-#
-# We do the vectorization with a ``TableVectorizer``. When we run the
-# cross-validation, the ``"baskets"`` table (which we marked as X) will be
-# split in train and test sets. To fit our ``TableVectorizer``, we want to use
-# only products that belong to a basket in the train set, not one in the test
-# set. So we do a semi-join to exclude any products that do not belong to a
-# basket that can be found in the ``"baskets"`` table currently being
-# processed.
-
-# %%
-
-# Note: using deferred or even defining a function is completely optional here,
-# because the computation it contains involves no control flow or assignments.
-# we only do it to simplify the computation graph by collapsing several
-# operations into a single function call.
-
-
-@skrub.deferred
```

Comment:
I agree this is redundant with the example 10. I wonder if there is another place we could illustrate the recommended practice of grouping steps together in `deferred` functions to simplify the graph avoid potential issues with pickling etc